extension/opampextension/internal/util/parse_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseDarwinDescription(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		input    string
+		expected string
+	}{
+		{
+			desc:     "basic use case",
+			input:    "ProductName: macOS\nProductVersion: 15.0.1\nBuildVersion: 24A348",
+			expected: "macOS 15.0.1",
+		},
+		{
+			desc:     "excessive white space",
+			input:    "			   \n	ProductName:		 macOS\nProductVersion: 15.0.1     \n \n	BuildVersion: 24A348\n",
+			expected: "macOS 15.0.1",
+		},
+		{
+			desc:     "random ordering & excessive white space",
+			input:    "\nProductVersion: 15.0.1     \n \n	BuildVersion: 24A348\n			   \n	ProductName:		 macOS",
+			expected: "macOS 15.0.1",
+		},
+		{
+			desc:     "blank input",
+			input:    "",
+			expected: "",
+		},
+		{
+			desc:     "missing ProductVersion",
+			input:    "ProductName: macOS\nBuildVersion: 24A348",
+			expected: "",
+		},
+		{
+			desc:     "missing ProductName",
+			input:    "ProductVersion: 15.0.1\nBuildVersion: 24A348",
+			expected: "",
```

Comment:
In cases where we can't parse the original input, could it be more helpful to return the original string? It seems like this would allow users to more easily report the unparsed value so that we can update the parsing logic if necessary. That said, I'm not familiar with the exact use case for this field so maybe the value would be problematic?

If we don't return the value, I think we should at least log an error that clearly indicates the value that could not be parsed.

Reply:
I'd rather not return the original input if it's "bad"/parsing fails, but I also do see the value in returning it anyways as it'd quickly indicate something is failing. It would also simplify things as the attribute would always be reported, regardless of parsing. Is it acceptable to report a bad/incorrect value here is what I'm not sure about. If so then I can change it to return the original if parsing fails